src/wallet.rs

```diff
@@ -634,26 +664,37 @@ impl Wallet {
             .map(|(_outpoint, amount, _sequence)| amount)
             .unwrap_or(Amount::ZERO);
 
-        let mut builder = self.bitcoin_wallet.build_tx();
-        builder
-            .ordering(bdk::wallet::tx_builder::TxOrdering::Untouched)
-            .add_recipient(op_drivechain.clone(), (ctip_amount + amount).to_sat())
-            .add_recipient(address_op_return, 0);
-
-        if let Some((ctip_outpoint, _, _)) = ctip {
-            let transaction_hex: String = self
+        let ctip_outpoint_with_tx = if let Some((ctip_outpoint, _, _)) = ctip {
+            let transaction_hex = self
                 .main_client
                 .get_raw_transaction(ctip_outpoint.txid, GetRawTransactionVerbose::<false>, None)
                 .await
                 .into_diagnostic()?;
+
             let ctip_outpoint = bdk::bitcoin::OutPoint {
                 txid: bdk::bitcoin::Txid::from_byte_array(ctip_outpoint.txid.to_byte_array()),
                 vout: ctip_outpoint.vout,
             };
-            let transaction_bytes = hex::decode(transaction_hex).unwrap();
-            let mut cursor = Cursor::new(transaction_bytes);
+
+            let transaction_bytes = hex::decode(transaction_hex).into_diagnostic()?;
+
             let transaction =
-                bdk::bitcoin::Transaction::consensus_decode(&mut cursor).into_diagnostic()?;
+                bdk::bitcoin::Transaction::consensus_decode(&mut Cursor::new(transaction_bytes))
+                    .into_diagnostic()?;
+
+            Some((ctip_outpoint, transaction))
+        } else {
+            None
+        };
+
+        let wallet = self.bitcoin_wallet.lock();
```

Comment:
You could drop this lock early, instead of holding it to the end of the function, across several blocking db calls.
Consider enabling Clippy lints relevant to Mutex/RwLock guards -
https://rust-lang.github.io/rust-clippy/stable/index.html#/significant_drop_in_scrutinee
https://rust-lang.github.io/rust-clippy/stable/index.html#/significant_drop_tightening